web/src/containers/BudgetSummary.js

```diff
@@ -0,0 +1,228 @@
+  </Fragment>
+);
+
+DataRowGroup.propTypes = {
+  data: PropTypes.object.isRequired
+};
+
+const HeaderRow = ({ title, numberCells }) => (
+  <tr>
+    <td className="bold">{title}</td>
+    {[...Array(numberCells)].map((_, i) => <td key={i} />)}
+  </tr>
+);
+
+HeaderRow.propTypes = {
+  title: PropTypes.string.isRequired,
+  numberCells: PropTypes.number
+};
+
+HeaderRow.defaultProps = {
+  numberCells: 12
+};
+
+const BudgetSummary = ({ data }) => (
+  <div className="py1 overflow-auto">
+    <table
+      className="h6 table-fixed table-bordered table-budget table-budget-summary"
+      style={{ minWidth: 1000 }}
+    >
+      <thead>
+        <tr>
+          <th style={{ width: 200 }} />
+          {YEAR_OPTIONS.map(yr => (
+            <th key={yr} className="bg-black white center" colSpan="3">
+              FFY {yr}
+            </th>
+          ))}
+          <th className="bg-black white center" colSpan="3">
+            Total
+          </th>
+        </tr>
+        <tr>
+          <th />
+          {[...Array(YEAR_OPTIONS.length + 1)].map((_, i) => (
+            <Fragment key={i}>
+              <th className="col-4">Total</th>
+              <th className="col-4">Federal share</th>
+              <th className="col-4">State share</th>
+            </Fragment>
+          ))}
+        </tr>
+      </thead>
+      <tbody className="bg-light-blue">
+        <HeaderRow title="HIT activities" />
+        <DataRowGroup data={data.HIT} />
+      </tbody>
+      <tbody className="bg-light-yellow">
+        <HeaderRow title="HIE activities" />
+        <DataRowGroup data={data.HIE} />
+      </tbody>
+      <tbody className="bg-light-green">
+        <HeaderRow title="MMIS activities" />
+        <DataRowGroup data={data.MMIS} />
+      </tbody>
+      <tbody>
+        <tr className="bold">
+          <td>Project total</td>
+          {Object.keys(data.combined).map(key => {
+            const val = data.combined[key];
+            return (
+              <Fragment key={key}>
+                <td className="mono right-align">{formatMoney(val)}</td>
+                <td className="mono right-align">{formatMoney(val * 0.9)}</td>
+                <td className="mono right-align">{formatMoney(val * 0.1)}</td>
+              </Fragment>
+            );
+          })}
+        </tr>
+      </tbody>
+    </table>
+  </div>
+);
+
+BudgetSummary.propTypes = {
+  data: PropTypes.object.isRequired
+};
+
+// merge and aggregate an array of objects
+// i.e., [{ a: 1, b: 2 }, { a: 3, b: 4 }] => { a: 4, b: 6 }
+const aggregateObjectArrays = data =>
+  data.reduce((obj, datum) => {
+    Object.keys(datum).forEach(key => {
+      obj[key] = (obj[key] || 0) + datum[key]; // eslint-disable-line no-param-reassign
+    });
+    return obj;
+  }, {});
+
+const getDataByActivityType = activities =>
+  ACTIVITY_TYPES.reduce((obj, id) => {
+    // filter by activity type (this isn't great because we'll be
+    // double-counting activities that belong to 2+ types
```

Comment:
I noticed this playing with the pr-deploy.  Glad to see this comment here.  I think we should open an issue to capture that this needs to be addressed.  I wonder if this is another level of detail in cost allocation...